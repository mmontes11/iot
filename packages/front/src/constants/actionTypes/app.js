export const TOGGLE_HAMBURGER_MENU = "TOGGLE_HAMBURGER_MENU";
export const TOGGLE_MAP_DIALOG = "TOGGLE_MAP_DIALOG";
export const SET_SHOW_ERROR = "SET_SHOW_ERROR";
