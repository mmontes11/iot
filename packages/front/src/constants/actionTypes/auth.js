export const IS_AUTH = "IS_AUTH";
export const SET_USERNAME = "SET_USERNAME";
export const SET_PASSWORD = "SET_PASSWORD";
export const LOGIN_REQUEST = "LOGIN_REQUEST";
export const LOGIN_REQUEST_SUCCESS = "LOGIN_REQUEST_SUCCESS";
export const LOGIN_REQUEST_ERROR = "LOGIN_REQUEST_ERROR";
