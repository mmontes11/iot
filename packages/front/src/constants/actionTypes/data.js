export const DATA_UPDATED = "DATA_UPDATED";
export const DATA_REQUEST = "DATA_REQUEST";
export const DATA_REQUEST_SUCCESS = "DATA_REQUEST_SUCCESS";
export const DATA_REQUEST_ERROR = "DATA_REQUEST_ERROR";
export const ADD_DATA_ITEM = "ADD_DATA_ITEM";
export const RESET_DATA = "RESET_DATA";
