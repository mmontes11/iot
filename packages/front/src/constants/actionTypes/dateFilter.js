export const DATE_FILTER_SELECTOR_TOGGLE = "DATE_FILTER_SELECTOR_TOGGLE";
export const DATE_FILTER_TIME_PERIOD_SELECT = "DATE_FILTER_TIME_PERIOD_SELECT";
export const DATE_FILTER_TIME_PERIOD_UPDATED = "DATE_FILTER_TIME_PERIOD_UPDATED";
export const DATE_FILTER_TIME_PERIODS_REQUEST = "DATE_FILTER_TIME_PERIODS_REQUEST";
export const DATE_FILTER_TIME_PERIODS_REQUEST_SUCCESS = "DATE_FILTER_TIME_PERIODS_REQUEST_SUCCESS";
export const DATE_FILTER_TIME_PERIODS_REQUEST_ERROR = "DATE_FILTER_TIME_PERIODS_REQUEST_ERROR";
export const DATE_FILTER_TIME_PERIODS_UPDATED = "DATE_FILTER_TIME_PERIODS_UPDATED";
export const DATE_FILTER_START_DATE_UPDATED = "DATE_FILTER_START_DATE_UPDATED";
export const DATE_FILTER_END_DATE_UPDATED = "DATE_FILTER_END_DATE_UPDATED";
