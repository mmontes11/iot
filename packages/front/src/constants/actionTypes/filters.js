export const FILTER_TYPE_SELECT = "FILTER_TYPE_SELECT";
export const ADD_FILTER_TYPE = "ADD_FILTER_TYPE";
export const DELETE_FILTER_TYPE = "DELETE_FILTER_TYPE";
