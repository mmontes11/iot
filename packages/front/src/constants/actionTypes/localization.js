export const UPDATE_LANGUAGE = "UPDATE_LANGUAGE";
