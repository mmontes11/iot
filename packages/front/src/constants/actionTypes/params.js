export const PARAM_SELECT = "PARAM_SELECT";
export const PARAM_UPDATE = "PARAM_UPDATE";
export const PARAM_RESET = "PARAM_RESET";
export const PARAM_REQUEST = "PARAM_REQUEST";
export const PARAM_REQUEST_SUCCESS = "PARAM_REQUEST_SUCCESS";
export const PARAM_REQUEST_ERROR = "PARAM_REQUEST_ERROR";
export const PARAM_ITEMS_UPDATED = "PARAM_ITEMS_UPDATED";
export const PARAM_DISABLE = "PARAM_DISABLE";
