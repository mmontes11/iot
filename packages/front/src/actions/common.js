import { RESET } from "constants/actionTypes/common";

export const reset = preserveError => dispatch => {
  dispatch({ type: RESET, preserveError });
};
