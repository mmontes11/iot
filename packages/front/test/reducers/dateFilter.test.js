import deepFreeze from "deep-freeze";
import dateFilterReducer from "reducers/dateFilter";
import {
  DATE_FILTER_SELECTOR_TOGGLE,
  DATE_FILTER_TIME_PERIOD_SELECT,
  DATE_FILTER_TIME_PERIOD_UPDATED,
  DATE_FILTER_TIME_PERIODS_REQUEST,
  DATE_FILTER_TIME_PERIODS_REQUEST_SUCCESS,
  DATE_FILTER_TIME_PERIODS_REQUEST_ERROR,
  DATE_FILTER_TIME_PERIODS_UPDATED,
  DATE_FILTER_START_DATE_UPDATED,
  DATE_FILTER_END_DATE_UPDATED,
} from "constants/actionTypes/dateFilter";
import { RESET } from "constants/actionTypes/common";
import { initialState } from "../constants/index";

const thingFilterInitialState = initialState.filters.dateFilter;
deepFreeze(thingFilterInitialState);

describe("reducers/dateFilter", () => {
  it("has a default state", () => {
    expect(dateFilterReducer(undefined, { type: "@init" })).toMatchSnapshot();
  });
  it("reduces nothing", () => {
    expect(dateFilterReducer(thingFilterInitialState, { type: "WHATEVER" })).toMatchSnapshot();
  });
  it("reduces DATE_FILTER_SELECTOR_TOGGLE", () => {
    expect(dateFilterReducer(thingFilterInitialState, { type: DATE_FILTER_SELECTOR_TOGGLE })).toMatchSnapshot();
  });
  it("reduces DATE_FILTER_TIME_PERIOD_SELECT", () => {
    expect(dateFilterReducer(thingFilterInitialState, { type: DATE_FILTER_TIME_PERIOD_SELECT })).toMatchSnapshot();
  });
  it("reduces DATE_FILTER_TIME_PERIOD_UPDATED", () => {
    expect(
      dateFilterReducer(thingFilterInitialState, { type: DATE_FILTER_TIME_PERIOD_UPDATED, updatedTimePeriod: "foo" }),
    ).toMatchSnapshot();
  });
  it("reduces DATE_FILTER_TIME_PERIODS_REQUEST", () => {
    expect(dateFilterReducer(thingFilterInitialState, { type: DATE_FILTER_TIME_PERIODS_REQUEST })).toMatchSnapshot();
  });
  it("reduces DATE_FILTER_TIME_PERIODS_REQUEST_SUCCESS", () => {
    expect(
      dateFilterReducer(thingFilterInitialState, { type: DATE_FILTER_TIME_PERIODS_REQUEST_SUCCESS }),
    ).toMatchSnapshot();
  });
  it("reduces DATE_FILTER_TIME_PERIODS_REQUEST_ERROR", () => {
    expect(
      dateFilterReducer(thingFilterInitialState, { type: DATE_FILTER_TIME_PERIODS_REQUEST_ERROR }),
    ).toMatchSnapshot();
  });
  it("reduces DATE_FILTER_TIME_PERIODS_UPDATED", () => {
    expect(
      dateFilterReducer(thingFilterInitialState, {
        type: DATE_FILTER_TIME_PERIODS_UPDATED,
        timePeriods: ["foo", "bar"],
      }),
    ).toMatchSnapshot();
  });
  it("reduces DATE_FILTER_START_DATE_UPDATED", () => {
    expect(
      dateFilterReducer(thingFilterInitialState, { type: DATE_FILTER_START_DATE_UPDATED, updatedStartDate: "foo" }),
    ).toMatchSnapshot();
  });
  it("reduces DATE_FILTER_END_DATE_UPDATED", () => {
    expect(
      dateFilterReducer(thingFilterInitialState, { type: DATE_FILTER_END_DATE_UPDATED, updatedEndDate: "bar" }),
    ).toMatchSnapshot();
  });
  it("reduces RESET", () => {
    expect(dateFilterReducer(thingFilterInitialState, { type: RESET })).toMatchSnapshot();
  });
});
