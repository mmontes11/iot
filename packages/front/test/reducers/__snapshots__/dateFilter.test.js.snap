// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/dateFilter has a default state 1`] = `
Object {
  "custom": Object {
    "endDate": null,
    "startDate": null,
  },
  "isCustomSelected": false,
  "timePeriod": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
}
`;

exports[`reducers/dateFilter reduces DATE_FILTER_END_DATE_UPDATED 1`] = `
Object {
  "custom": Object {
    "endDate": "bar",
    "startDate": null,
  },
  "isCustomSelected": false,
  "timePeriod": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
}
`;

exports[`reducers/dateFilter reduces DATE_FILTER_SELECTOR_TOGGLE 1`] = `
Object {
  "custom": Object {
    "endDate": null,
    "startDate": null,
  },
  "isCustomSelected": true,
  "timePeriod": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
}
`;

exports[`reducers/dateFilter reduces DATE_FILTER_START_DATE_UPDATED 1`] = `
Object {
  "custom": Object {
    "endDate": null,
    "startDate": "foo",
  },
  "isCustomSelected": false,
  "timePeriod": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
}
`;

exports[`reducers/dateFilter reduces DATE_FILTER_TIME_PERIOD_SELECT 1`] = `
Object {
  "custom": Object {
    "endDate": null,
    "startDate": null,
  },
  "isCustomSelected": false,
  "timePeriod": Object {
    "isActive": true,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
}
`;

exports[`reducers/dateFilter reduces DATE_FILTER_TIME_PERIOD_UPDATED 1`] = `
Object {
  "custom": Object {
    "endDate": null,
    "startDate": null,
  },
  "isCustomSelected": false,
  "timePeriod": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": "foo",
  },
}
`;

exports[`reducers/dateFilter reduces DATE_FILTER_TIME_PERIODS_REQUEST 1`] = `
Object {
  "custom": Object {
    "endDate": null,
    "startDate": null,
  },
  "isCustomSelected": false,
  "timePeriod": Object {
    "isActive": false,
    "isDisabled": true,
    "isLoading": true,
    "items": Array [],
    "selectedItem": null,
  },
}
`;

exports[`reducers/dateFilter reduces DATE_FILTER_TIME_PERIODS_REQUEST_ERROR 1`] = `
Object {
  "custom": Object {
    "endDate": null,
    "startDate": null,
  },
  "isCustomSelected": false,
  "timePeriod": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
}
`;

exports[`reducers/dateFilter reduces DATE_FILTER_TIME_PERIODS_REQUEST_SUCCESS 1`] = `
Object {
  "custom": Object {
    "endDate": null,
    "startDate": null,
  },
  "isCustomSelected": false,
  "timePeriod": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
}
`;

exports[`reducers/dateFilter reduces DATE_FILTER_TIME_PERIODS_UPDATED 1`] = `
Object {
  "custom": Object {
    "endDate": null,
    "startDate": null,
  },
  "isCustomSelected": false,
  "timePeriod": Object {
    "isActive": true,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [
      "foo",
      "bar",
    ],
    "selectedItem": null,
  },
}
`;

exports[`reducers/dateFilter reduces RESET 1`] = `
Object {
  "custom": Object {
    "endDate": null,
    "startDate": null,
  },
  "isCustomSelected": false,
  "timePeriod": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
}
`;

exports[`reducers/dateFilter reduces nothing 1`] = `
Object {
  "custom": Object {
    "endDate": null,
    "startDate": null,
  },
  "isCustomSelected": false,
  "timePeriod": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
}
`;
